aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -2049,6 +2049,14 @@ std::tuple<Tensor&, Tensor&> linalg_eigh_out(const Tensor& input, std::string up
 }
 
 Tensor linalg_eigvalsh(const Tensor& input, std::string uplo) {
+  // if input requires grad we must compute the eigenvectors to make this function differentiable
+  // the eigenvectors are not exposed to the user
+  if (torch::autograd::compute_requires_grad(input)) {
+    Tensor values;
+    std::tie(values, std::ignore) = at::linalg_eigh(input, uplo);
```

Comment:
nit. Perhaps it's simpler to just `return std::get<0>(at::linalg_eig(input, uplo))`?

Reply:
I find the current way more clear: it's easy to understand from the names of the return tensors that we are saving the eigenvalues and ignoring the eigenvectors. `return std::get<0>` is valid, but hides bits of information.